### ISSUE TITLE
Move embeddings & WikipediaSimilaritySearchTool from API to FW

### ISSUE THREAD

**Issue by michael-desmond** [i-am-bee/bee-agent-framework#166]
**Is your feature request related to a problem? Please describe.**
The WikipediaSimilaritySearchTool and the related embedding capabilities are very useful when working with smaller LLMs. For example with granite bee its really the only way to use the Wikipedia tool. Granite will eventually get a larger context window but the ability to chunk/embed/retrieve seems generally useful for tool developers, and provides some core building blocks for optimized tools. 

Right now if I want to provide a granite bee + wikipedia example in the FW I cant use the WikipediaSimilaritySearchTool.

**Describe the solution you'd like**
Move embeddings & WikipediaSimilaritySearchTool from API to FW to make the functionality available at lower level in stack. 

**Describe alternatives you've considered**
Summarization of wikipedia article using an llm post processing step. 

**Additional context**
None.

**Comment by Tomas2D** [comment]
We would need to extend the `BaseLLM` class to introduce the embeddings by adding a new method (similar to `tokenize`). This can be done in a separate issue. It would do that instead of bringing new classes in.
Regarding [WikipediaSimilaritySearchTool](https://github.com/i-am-bee/bee-api/blob/3a7448a31ea44816d107bb5783b6428219f8bfde/src/runs/execution/tools/wikipedia-similarity-search-tool.ts#L34), I would create a general tool instead. That tool would wrap up the provided tool, and that's it.

**Comment by Tomas2D** [comment]
Such a tool can now be created by using the `pipe`/`extend` method available on a tool instance.
Example: https://github.com/i-am-bee/bee-agent-framework/blob/main/examples/tools/custom/piping.ts

However the functionality is still blocked by #176

**Comment by Tomas2D** [comment]
Embedding has been added and functionality has been refactored in the API.
